onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu

```diff
@@ -82,18 +82,23 @@ template <typename T, unsigned TPB, bool Simplified>
 __global__ void SkipLayerNormKernel(
     const int ld, const T* input, const T* skip,
     const T* beta, const T* gamma, const T* bias,
-    const T epsilon, T* output, T* skip_input_bias_add_output) {
+    const T epsilon, T* output, T* skip_input_bias_add_output, const bool skip_broadcasted, int skip_size) {
   const T reverse_ld = T(1.f / ld);
   const int offset = blockIdx.x * ld;
 
   KeyValuePairSum pair_sum;
   // reduce x and x^2
   cub::KeyValuePair<T, T> thread_data(0, 0);
 
+
   for (int i = threadIdx.x; i < ld; i += TPB) {
     const int idx = offset + i;
 
     const T val = (bias == nullptr) ? input[idx] + skip[idx] : input[idx] + skip[idx] + bias[i];
+
+    if (skip_broadcasted){
```

Comment:
you can do something like
const T skip_data = skip_broadcasted ? skip[idx % skip_size] : skip[idx]
and then use skip_data